lib/server/services/lhcPeriod/LhcPeriodStatisticsService.js

```diff
@@ -46,6 +47,21 @@ class LhcPeriodStatisticsService {
         return data ? lhcPeriodStatisticsAdapter.toEntity(data) : null;
     }
 
+    /**
+     * Find a lhc period model by its name or id including its statistics
+     * @param {LhcPeriodIdentifier} identifier the criteria to find run type
+     * @throws {NotFoundError} in case there is no LHC Period with given identifier
+     * @return {Promise<LhcPeriodStatistics[]>} the run type found or null
+     */
+    async getOneOrFail({ id, name }) {
+        const lhcPeriodStatistics = await this.getByIdentifier({ id, name });
+        if (!lhcPeriodStatistics) {
+            const criteriaExpression = id ? `id (${id})` : `name (${name})`;
```

Comment:
Be really careful when using ternary with number! Here if `id = 0` you will expect a name and have weird errors!

Reply:
Same thing in the `getByIdentifier`

Reply:
right, applied